accord-core/src/main/java/accord/local/Command.java

```
@@ -18,1178 +18,1435 @@
-            throw new AssertionError("Invalid response from validate function");
-
-        setExecuteAt(executeAt);
-        setPromised(ballot);
-        setAccepted(ballot);
-
-        // TODO (desired, clarity/efficiency): we don't need to set the route here, and perhaps we don't even need to
-        //  distributed partialDeps at all, since all we gain is not waiting for these transactions to commit during
-        //  recovery. We probably don't want to directly persist a Route in any other circumstances, either, to ease persistence.
-        set(safeStore, coordinateRanges, acceptRanges, shard, route, null, Ignore, partialDeps, Set);
+        @Override
+        public boolean equals(Object o)
+        {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Command command = (Command) o;
+            return txnId.equals(command.txnId())
+                    && status == command.saveStatus()
+                    && durability == command.durability()
+                    && Objects.equals(homeKey, command.homeKey())
+                    && Objects.equals(progressKey, command.progressKey())
+                    && Objects.equals(route, command.route())
+                    && Objects.equals(promised, command.promised())
+                    && listeners.equals(command.listeners());
+        }
 
-        // set only registers by transaction keys, which we mightn't already have received
-        if (!known().isDefinitionKnown())
-            safeStore.register(keys, acceptRanges, this);
+        @Override
+        public String toString()
+        {
+            return "Command@" + System.identityHashCode(this) + '{' + txnId + ':' + status + '}';
+        }
 
-        setStatus(Accepted);
-        safeStore.progressLog().accepted(this, shard);
-        notifyListeners(safeStore);
+        @Override
+        public int hashCode()
```

maybe throw `UnsupportedOperationException`, since we don't expect this method to be used?